src/dnaio/_core.pyx

```diff
@@ -384,15 +385,16 @@ def fastq_iter(file, sequence_class, Py_ssize_t buffer_size):
                 raise FastqFormatError(
                     "Length of sequence and qualities differ", line=n_records * 4 + 3)
 
-            ### Copy record into python variables
-            # PyUnicode_DecodeLatin1 is 50% faster than PyUnicode_DecodeASCII.
-            # This is because PyUnicode_DecodeLatin1 is an alias for
-            # _PyUnicode_FromUCS1. Which directly copies the bytes into a
-            # string object after some checks. With PyUnicode_DecodeASCII,
-            # there is an extra check whether characters exceed 128.
-            name = PyUnicode_DecodeLatin1(c_buf + name_start, name_length, 'strict')
-            sequence = PyUnicode_DecodeLatin1(c_buf + sequence_start, sequence_length, 'strict')
-            qualities = PyUnicode_DecodeLatin1(c_buf + qualities_start, qualities_length, 'strict')
+            # Constructing objects with PyUnicode_New and memcpy bypasses some of
+            # the checks otherwise done when using PyUnicode_DecodeLatin1 or similar
+            name = PyUnicode_New(name_length, 255)
+            sequence = PyUnicode_New(sequence_length, 255)
+            qualities = PyUnicode_New(qualities_length, 255)
+            if <PyObject*>name == NULL or <PyObject*>sequence == NULL or <PyObject*>qualities == NULL:
+                raise MemoryError()
+            memcpy(PyUnicode_1BYTE_DATA(name), c_buf + name_start, name_length)
```

Comment:
Apparently [PyUnicode_DATA](https://docs.python.org/3.11/c-api/unicode.html#c.PyUnicode_DATA) will also work as a `void *`. Maybe it is faster?

Reply:
I cannot find its implementation, but `PyUnicode_1BYTE_DATA` is just a macro, so I don’t think this takes any relevant time at all.

Reply:
You are right: https://github.com/python/cpython/blob/45f5f52601ebccb195c19cb0a77beaf7f7dfa56a/Include/cpython/unicodeobject.h#L320